automount-actions/scripts/bin/auto-install.sh

```diff
@@ -76,38 +74,41 @@ validate_assertion_file() {
         return 1
     }
     
-    # Check for required assertion types
-    _has_account_key=false
-    _has_snap_declaration=false  
-    _has_snap_revision=false
+    # Use permission-style integer flags to track which assertions are found
+    # This allows us to determine exactly which ones are missing based on the sum
+    _has_account_key=0
+    _has_snap_declaration=0
+    _has_snap_revision=0
     
     # Read the file and look for the required assertion types
-    while read -r _line; do
-        case "$_line" in
-            "type: account-key"*)     _has_account_key=true ;;
-            "type: snap-declaration"*) _has_snap_declaration=true ;;
-            "type: snap-revision"*)   _has_snap_revision=true ;;
+    # Parse key-value pairs properly by splitting on ': '
+    while IFS=': ' read -r _key _val; do
+        [ "$_key" = "type" ] || continue
+        case "$_val" in
+            "account-key")     _has_account_key=1     ;;
+            "snap-declaration") _has_snap_declaration=2 ;;
+            "snap-revision")   _has_snap_revision=4   ;;
         esac
     done < "$_assert_file"
     
-    # Verify all three required assertion types are present
-    if ! $_has_account_key; then
-        log "WARN: Assertion file missing account-key: $_assert_file"
-        return 1
-    fi
+    # Calculate sum to determine which assertions are present
+    _sum=$(( _has_account_key + _has_snap_declaration + _has_snap_revision ))
     
-    if ! $_has_snap_declaration; then
-        log "WARN: Assertion file missing snap-declaration: $_assert_file"
-        return 1
-    fi
+    # If sum is 7 (1+2+4), all assertions are present
+    [ "$_sum" -eq 7 ] && {
+        log "VALIDATION: Assertion file contains all required types: $_assert_file"
+        return 0
+    }
     
-    if ! $_has_snap_revision; then
-        log "WARN: Assertion file missing snap-revision: $_assert_file"
-        return 1
-    fi
+    # Build missing assertions message based on sum
+    _missing=""
+    [ $(( _sum & 1 )) -eq 0 ] && _missing="${_missing} account-key"
+    [ $(( _sum & 2 )) -eq 0 ] && _missing="${_missing} snap-declaration" 
+    [ $(( _sum & 4 )) -eq 0 ] && _missing="${_missing} snap-revision"
     
-    log "VALIDATION: Assertion file contains all required types: $_assert_file"
-    return 0
+    log "WARN: Assertion file missing${_missing}: $_assert_file"
+    
+    return 1
```

Comment:
The more I think about it, the less I like this -- it's a bit more opaque because it's edging on Clever Programmer :tm:, and I don't think it even efficiently works without `eval` (which, even though even ancient pdksh supports, isn't POSIX).

At the very least, the point of doing it this way is to condense the code and reduce the almighty SLOC. I probably revert this, unless @atomcult can unbreak my brain on it.